src/olympia/addons/tasks.py

```diff
@@ -125,7 +127,7 @@ def delete_all_addon_media_with_backup(id, **kwargs):
 def restore_all_addon_media_from_backup(id, **kwargs):
     addon = Addon.unfiltered.all().get(pk=id)
     disabled_addon_content = DisabledAddonContent.objects.filter(addon=addon).last()
```

Comment:
Could approve efficiency when backup_storage is not enabled by only doing this query if it's enabled.
`qs = DisabledAddonContent.objects.filter(addon=addon)`
and then `if backup_storage_enabled() and (disabled_addon_content := qs.last()):`

(if we care about this case)